lib/client-api/src/inferControls.ts

```diff
@@ -10,51 +10,50 @@ type ControlsMatchers = {
 };
 
 const inferControl = (argType: ArgType, name: string, matchers: ControlsMatchers): any => {
-  const { type } = argType;
-  if (!type) {
-    return null;
+  const { type, options } = argType;
+  if (!type && !options) {
+    return undefined;
   }
 
   // args that end with background or color e.g. iconColor
   if (matchers.color && matchers.color.test(name)) {
-    return { type: 'color' };
+    return { control: { type: 'color' } };
   }
 
   // args that end with date e.g. purchaseDate
   if (matchers.date && matchers.date.test(name)) {
-    return { type: 'date' };
+    return { control: { type: 'date' } };
   }
 
   switch (type.name) {
     case 'array': {
       const { value } = type;
       if (value?.name && ['object', 'other'].includes(value.name)) {
         return {
-          type: 'object',
-          validator: (obj: any) => Array.isArray(obj),
+          control: {
+            type: 'object',
+            validator: (obj: any) => Array.isArray(obj),
```

Comment:
is `validator` documented?

Reply:
LOL almost zero memory of this 😭 

Per our slack discussion, let's get rid of validators and also use the JSON editor UI for **ALL** arrays moving forward